prompt-service/src/unstract/prompt_service/helper.py

```diff
@@ -260,7 +260,9 @@ def construct_and_run_prompt(
     metadata: dict[str, Any],
 ) -> str:
     platform_postamble = tool_settings.get(PSKeys.PLATFORM_POSTAMBLE, "")
-    if tool_settings.get(PSKeys.SUMMARIZE_AS_SOURCE):
+    summarize_as_source = tool_settings.get(PSKeys.SUMMARIZE_AS_SOURCE)
+    enable_highlight = tool_settings.get(PSKeys.ENABLE_HIGHLIGHT, False)
+    if not enable_highlight or summarize_as_source:
```

Comment:
@Deepak-Kesavan why is this `not  enable_highlight` added?

Reply:
No need to extract line numbers if highlight is disabled. That is why it is added.